src/core/lv_obj_tree.c

```diff
@@ -416,14 +418,16 @@ int32_t lv_obj_get_index(const lv_obj_t * obj)
     LV_ASSERT_OBJ(obj, MY_CLASS);
 
     lv_obj_t * parent = lv_obj_get_parent(obj);
-    if(parent == NULL) return 0xFFFFFFFF;
+    if(parent == NULL) return -1;
 
     int32_t i = 0;
     for(i = 0; i < parent->spec_attr->child_cnt; i++) {
         if(parent->spec_attr->children[i] == obj) return i;
     }
 
-    return -1; /*Shouldn't happen*/
+    /*Shouldn't reach this point*/
+    LV_ASSERT(0);
```

Comment:
We shouldn't reach this point, that's why I've added the `LV_ASSERT(0)`, which will always fails if reached.

Reply:
Assert could be disabled... thus we should add return -1 at the end.

Reply:
Yep, that's why the return statement was kept